tests/integration/ambient/wasm_test.go

```diff
@@ -0,0 +1,231 @@
+		"TargetWaypointName": constants.DefaultNamespaceWaypoint,
+	}
+
+	if len(imagePullPolicy) != 0 {
+		args["ImagePullPolicy"] = imagePullPolicy
+	}
+
+	if err := applyWasmConfig(ctx, apps.Namespace.Name(), args, path); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func uninstallWasmExtension(ctx framework.TestContext, pluginName, path string) error {
+	args := map[string]any{
+		"WasmPluginName": pluginName,
+	}
+	if err := ctx.ConfigIstio().EvalFile(apps.Namespace.Name(), args, path).Delete(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func sendTraffic(ctx framework.TestContext, checker echo.Checker, options ...retry.Option) {
+	ctx.Helper()
+	if len(GetClientInstances()) == 0 {
+		ctx.Fatal("there is no client")
+	}
+	cltInstance := GetClientInstances()[0]
+
+	defaultOptions := []retry.Option{retry.Delay(100 * time.Millisecond), retry.Timeout(200 * time.Second)}
+	httpOpts := echo.CallOptions{
+		To: GetTarget(),
+		Port: echo.Port{
+			Name: "http",
+		},
+		HTTP: echo.HTTP{
+			Path:   "/path",
+			Method: "GET",
+		},
+		Count: 1,
+		Retry: echo.Retry{
+			Options: append(defaultOptions, options...),
+		},
+		Check: checker,
+	}
+
+	_ = cltInstance.CallOrFail(ctx, httpOpts)
+}
+
+func sendTrafficToHostname(ctx framework.TestContext, checker echo.Checker, hostname string, options ...retry.Option) {
+	ctx.Helper()
+	if len(GetClientInstances()) == 0 {
+		ctx.Fatal("there is no client")
+	}
+	cltInstance := GetClientInstances()[0]
+
+	defaultOptions := []retry.Option{retry.Delay(100 * time.Millisecond), retry.Timeout(200 * time.Second)}
+	httpOpts := echo.CallOptions{
+		Address: hostname,
+		Port: echo.Port{
+			Name:        "http",
+			ServicePort: 80,
+			Protocol:    protocol.HTTP,
+		},
+		HTTP: echo.HTTP{
+			Path:    "/path",
+			Method:  "GET",
+			Headers: headers.New().WithHost(fmt.Sprintf("%s.com", GetTarget().ServiceName())).Build(),
+		},
+		Count: 1,
+		Retry: echo.Retry{
+			Options: append(defaultOptions, options...),
+		},
+		Check: checker,
+	}
+
+	_ = cltInstance.CallOrFail(ctx, httpOpts)
+}
+
+func TestWasmPluginConfigurations(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(t framework.TestContext) {
+			testCases := []struct {
+				desc         string
+				name         string
+				testHostname string
+				yamlFilePath string
+			}{
+				{
+					desc:         "initial creation with latest for a gateway",
+					name:         "gateway-wasm-test",
+					testHostname: fmt.Sprintf("%s-gateway-istio.%s.svc.cluster.local", GetTarget().ServiceName(), apps.Namespace.Name()),
+					yamlFilePath: "testdata/gateway-wasm-filter.yaml",
+				},
+				{
+					desc:         "Configure WebAssembly filter for waypoint gateway",
+					name:         "waypoint-wasm-test",
+					yamlFilePath: "testdata/wasm-filter.yaml",
+				},
+				{
+					desc:         "Configure WebAssembly filter for specific service",
+					name:         "service-wasm-test",
+					yamlFilePath: "testdata/wasm-filter-service.yaml",
```

Comment:
here the name can be service-wasm-filter.yaml so that all have consistent naming